profile/profiler.go

```diff
@@ -82,6 +82,9 @@ func NewProfiler(executionTime float64, printInterval uint64, vmNum, level int,
 		}
 		profiler.cmd.Args = append(profiler.cmd.Args, "--core", core)
 	} else {
+		if socket > -1 {
```

Comment:
what is this check? it's not intuitive. It is unclear what this logic does and why

Reply:
If *bindSocket is set, the profiler will just measure the socket.

Reply:
set to what? it might be set to -1

Reply:
you need to make the code clear, do not assume that people would remember this later.

Reply:
still missing a comment

Reply:
Done